lib/flapjack/gateways/pager_duty.rb

```diff
@@ -184,26 +185,26 @@ def start
             Kernel.sleep(10)
           end
 
-          # begin
-
-            Zermelo.redis = Flapjack.redis
-
-            loop do
-              @lock.synchronize do
-                # ensure we're the only instance of the PagerDuty acknowledgement check running (with a naive
-                # timeout of five minutes to guard against stale locks caused by crashing code) either in this
-                # process or in other processes
-                if Flapjack.redis.setnx(SEM_PAGERDUTY_ACKS_RUNNING, 'true') == 0
-                  Flapjack.logger.debug("skipping looking for acks in PagerDuty as this is already happening")
-                else
-                  Flapjack.redis.expire(SEM_PAGERDUTY_ACKS_RUNNING, 300)
-                  find_pagerduty_acknowledgements
-                  Flapjack.redis.del(SEM_PAGERDUTY_ACKS_RUNNING)
-                end
+        # begin
+
+          Zermelo.redis = Flapjack.redis
+
+          loop do
+            @lock.synchronize do
+              # ensure we're the only instance of the PagerDuty acknowledgement check running (with a naive
+              # timeout of five minutes to guard against stale locks caused by crashing code) either in this
```

Comment:
As does the logic :( https://github.com/flapjack/zermelo/blob/master/lib/zermelo/locks/redis_lock.rb is used in `zermelo` for class-level locking, we could maybe use that class directly to set up something safer, with an appropriate timeout (it should release the lock even on exception, but nothing's going to protect against, e.g. `kill -9`)).